File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.cpp
class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int n = nums1.size();
        int minimum = INT_MAX;
        bool allEven = true;

        for (int i=0;i<n;i++){
            minimum = min(minimum,nums1[i]);

            if(nums1[i]%2 !=0){
                allEven = false;
            }
        }

        if(minimum%2 !=0){
            return true;
        }
        if(allEven){
            return true;
        }
        return false;
    }
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.py
class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        n = len(nums1)
        minimum = nums1[0]
        allEven = True

        for x in range(n):
            minimum = min(minimum,nums1[x])

            if nums1[x] % 2 != 0:
                allEven = False

        if minimum % 2 !=0:
            return True
        if allEven:
            return True
        
        return False